clio-server/src/main/scala/org/broadinstitute/clio/server/service/IndexService.scala

```diff
@@ -45,26 +45,49 @@ abstract class IndexService[CI <: ClioIndex](
     metadata: clioIndex.MetadataType,
     force: Boolean = false
   ): Source[UpsertId, NotUsed] = {
-    // query by key to see if the document already exists
-    if (!force) {
-      queryMetadataForKey(indexKey)
-        .fold[Either[Exception, clioIndex.MetadataType]](
-          // If the query doesn't return a document and the document status is null then set it to the default value.
-          Right(
-            setDefaultDocumentStatus(metadata)
-          )
-        ) { (_, existingMetadataJson) =>
-          // validation will check to make sure no fields are being overwritten
-          validateUpsert(existingMetadataJson, metadata)
-        }
-        .flatMapConcat[UpsertId, NotUsed] {
-          case Right(updatedMetadata) =>
-            upsertToStream(indexKey, updatedMetadata)
-          case Left(rejection) => Source.failed(rejection)
+    // query by key to see if the document already exists if force is false
+    val validateOrError =
+      if (!force) {
+        queryMetadataForKey(indexKey)
```

Comment:
Akka streams are lazy & pluggable, so I think we could extract the query -> fold -> decoding portion into a local `val`. Then do something like:
```scala
val theQuery = queryMetadataForKey(indexKey).fold(...) { existingMetadataJson.as[Metadata] }
val validateOrError =
  if (!force) {
    theQuery.map { existingOrErr =>
      for {
        existing <- existingOrErr
        ... validate ...
      } yield {
        ...
      }
    }
  } else {
    metadata.documentStatus.fold(theQuery)(_ => Source.single(...))
  }
```